server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -216,19 +220,31 @@ private synchronized void updateRemoteClusters(Map<String, Tuple<String, List<Tu
                 List<Tuple<String, Supplier<DiscoveryNode>>> seedList = entry.getValue().v2();
                 String proxyAddress = entry.getValue().v1();
 
-                RemoteClusterConnection remote = this.remoteClusters.get(entry.getKey());
+                String clusterAlias = entry.getKey();
+                RemoteClusterConnection remote = this.remoteClusters.get(clusterAlias);
+                ConnectionProfile connectionProfile = this.remoteClusterConnectionProfiles.get(clusterAlias);
                 if (seedList.isEmpty()) { // with no seed nodes we just remove the connection
                     try {
                         IOUtils.close(remote);
                     } catch (IOException e) {
-                        logger.warn("failed to close remote cluster connections for cluster: " + entry.getKey(), e);
+                        logger.warn("failed to close remote cluster connections for cluster: " + clusterAlias, e);
                     }
-                    remoteClusters.remove(entry.getKey());
+                    remoteClusters.remove(clusterAlias);
                     continue;
                 }
 
                 if (remote == null) { // this is a new cluster we have to add a new representation
-                    String clusterAlias = entry.getKey();
+                    remote = new RemoteClusterConnection(settings, clusterAlias, seedList, transportService, numRemoteConnections,
+                        getNodePredicate(settings), proxyAddress);
+                    remoteClusters.put(clusterAlias, remote);
+                } else if (connectionProfileChanged(remote.getConnectionManager().getConnectionProfile(), connectionProfile)) {
+                    // New ConnectionProfile. Must tear down existing connection
```

Comment:
if I understand this correctly we will cause existing connections to fail and further, if concurrently somebody is using is it will not be able to retry since it's close and there will be intermediate failures in such a case. I think in this case we have to somehow reconfigure instead of closing? I know I would like this to be simple but the sideeffects of a close of a connection while some consumer is "in the middle of something" might cause a lot of headaches since it's a rare event, hard to debug and to test. WDYT?

Reply:
What you're describing is possible but will require changes at multiple levels of the networking stack. The `ConnectionProfile` is not currently mutable. So compression and ping intervals will need to be mutable. It also raises the question of whether connection timeout and handshake timeout would be mutable. I guess not since they are not dynamic settings currently. I think it would be very complicated to change the number of channels and their types, so that should stay static.

The compression change would be straightforward. For every request it we could just read from the volatile variable in the connection profile. Ping interval would be trickier because we would need to reschedule it to a different ping interval (probably at the completion of a ping-run).

It can be done, but it is uncertain to make the next release. It also would require reworking a lot with `ConnectionProfile`. Let me know if this is your preference and I will pursue updating this PR with those changes at some point in the future.